include/nlohmann/detail/conversions/to_json.hpp

```diff
@@ -443,12 +443,8 @@ inline void to_json(BasicJsonType& j, const T& t)
 template<typename BasicJsonType>
 inline void to_json(BasicJsonType& j, const std_fs::path& p)
 {
-#ifdef JSON_HAS_CPP_20
-    const std::u8string s = p.u8string();
+    const auto& s = p.u8string();
```

Comment:
Removing the conditional compilation adds an unnecessary copy to those using it in C++17.

Reply:
Unfortunately, without this change the tests fail because `BasicJsonType::is_string` no longer holds in corresponding `from_json`.

Reply:
Testing against `201811L` seems to solve the issue.

Reply:
@sergiud Oh, that's annoying.  So, libc++ in Clang 11.0.x sets `__cpp_lib_char8_t 201811L` but doesn't actually implement `p.u8string()` returning `std::u8string` until it's set to `201907L`?

That's ugly.

However, this still means that everything using C++17 has to pay for this copy unnecessarily, where they didn't before.  Would the following work?

```
    const auto& s = p.u8string();
#if defined(__cpp_lib_char8_t) && (__cpp_lib_char8_t >= 201907L)
    j = std::string(s.begin(), s.end());
#else
    j = s; // returns std::string in C++17
#endif
```

Reply:
Yes, `__cpp_lib_char8_t` version is somewhat unreliable. Therefore, instead of feature testing against a specific version for `char8_t` library support a better approach is probably to provide a `std::u8string` overload whenever `__cpp_lib_char8_t` is defined.

Reply:
Looks reasonable.